### PR TITLE
Add _foreach_{mul/div}.{ScalarList/Tensor} and their variants

### PR DESCRIPTION
**Pull request by CuiYifeng**
- [x] _foreach_mul.Tensor
- [x] \_foreach_mul_.Tensor
- [x] _foreach_div.Tensor
- [x] \_foreach_div_.Tensor
- [x] _foreach_mul.ScalarList
- [x] \_foreach_mul_.ScalarList
- [x] _foreach_div.ScalarList
- [x] \_foreach_div_.ScalarList

### CODE REVIEW

**Comment by xytintel** [comment]
Pls also add `_foreach_mul.ScalarList`, `_foreach_mul_.ScalarList`, `_foreach_div.ScalarList` and `_foreach_div_.ScalarList`.

**Comment by xytintel** [comment]
Why use the name `BinaryOpTensor` instead of `BinaryOpScalarTensor` ?

File: src/ATen/native/xpu/sycl/ForeachBinaryOpTensorKernels.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+#include <ATen/ops/empty_like_native.h>
+
+namespace at::native::xpu {
+
+template <typename T, template <class> class Op>
+std::vector<Tensor> foreach_binary_op(
+    TensorList tensors,
+    const Tensor& scalar,
+    const Scalar& alpha = 1) {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/ForeachBinaryOpTensorKernels.cpp:18]: Remove `alpha` if not need

File: src/ATen/native/xpu/sycl/ForeachBinaryOpTensorKernels.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      Op<opmath_t>(),
+      scalar.data_ptr<T>());
+  return tensor_lists[1];
+}
+
+template <typename T, template <class> class Op>
+void foreach_binary_op_(
+    TensorList tensors,
+    const Tensor& scalar,
+    const Scalar& alpha = 1) {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/ForeachBinaryOpTensorKernels.cpp:59]: ditto.

**Comment by CuiYifeng** [comment]
> Why use the name `BinaryOpTensor` instead of `BinaryOpScalarTensor` ?

Thanks for the reminder. Changes done.

**Pull request merged**